Python/compile.c

```diff
@@ -3357,8 +3357,11 @@ compiler_assert(struct compiler *c, stmt_ty s)
 
     if (c->c_optimize)
         return 1;
-    if (s->v.Assert.test->kind == Tuple_kind &&
-        asdl_seq_LEN(s->v.Assert.test->v.Tuple.elts) > 0)
+    if ((s->v.Assert.test->kind == Tuple_kind &&
+        asdl_seq_LEN(s->v.Assert.test->v.Tuple.elts) > 0) ||
+        (s->v.Assert.test->kind == Constant_kind &&
+         PyTuple_Check(s->v.Assert.test->v.Constant.value) &&
+         PyTuple_Size(s->v.Assert.test->v.Constant.value) > 0)) 
```

Comment:
By moving the `if (c->c_optimize)` check after these checks, we can warn even when `__debug__ == False`

Reply:
Done.